src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -162,12 +162,15 @@ struct PerfRecordSample {
   if (flags.sample_type & PERF_SAMPLE_STACK_USER) {
     ring_buffer->ReadRawAtOffset(&event.stack_size, current_offset, sizeof(uint64_t));
     current_offset += sizeof(uint64_t);
-    event.stack_data = make_unique_for_overwrite<uint8_t[]>(event.stack_size);
+    if (event.stack_size != 0u) {
```

Comment:
Nit: maybe add a comment that the dyn_size is after the actual stack but we don't want to copy the "whole stack".

Reply:
Should be done!